### PR TITLE
Mention "improve software you rely" as a reason to contribute.

### PR DESCRIPTION
**Pull request by Zimmi48**
- [x] Have you followed the [contributing guidelines](https://github.com/github/opensource.guide/blob/master/CONTRIBUTING.md)?
- [x] Have you explained what your changes do, and why they add value to the Guides?

-----

It feels to me like this is probably the single most important reason why people contribute to open source. Maybe I'm wrong about the priority order, but it would be weird if this reason was not mentioned at all.

Obviously, I'm open to moving the sub-section down (for now I added it on top of the list), and to rephrasing it entirely.

### CODE REVIEW

**Comment by Zimmi48** [comment]
Maybe I should add a sentence saying that if you don't know how to write the patch, reporting a bug is also a good way to give a chance to your bug getting fixed (to acknowledge that contributing is not just about writing code). Let me know what you think.

File: _articles/how-to-contribute.md [hunk 1]
```diff
@@ -31,6 +31,10 @@ Contributing to open source can be a rewarding way to learn, teach, and build ex
 
 Why do people contribute to open source? Plenty of reasons!
 
+### Improve software you rely on
+
+Lots of open source contributors start by being users of software they contribute to. When you find a bug, if you are able to write a patch fixing it, you could keep it for yourself, but contributing it back is a best way to ensure that your friends (and yourself when you update to the next release) will be able to benefit from it.
```
Comment (mlinksva) [inline _articles/how-to-contribute.md]: The second sentence is pretty long, I wonder if it could be tightened or broken up?

**Comment by mlinksva** [comment]
Requested an edit to make more readable, but I think that's right and a good addition, aligns well with later the article https://opensource.guide/how-to-contribute/#finding-a-project-to-contribute-to

> Instead, start by thinking about the projects you already use, or want to use. The projects you’ll actively contribute to are the ones you find yourself coming back to.

**Comment by Zimmi48** [comment]
Thanks @mlinksva! I've split the long sentence in two. It looks much better to me now. Let me know what you think.

**Comment by mlinksva** [comment]
Thanks @Zimmi48! I think this subsection might be further improved by not focusing only on bug fixes (elsewhere we emphasize there are other ways to contribute, ie to improve the software) but I'll go ahead and merge as that could be another improvement.

**Pull request merged**